envoy/upstream/cluster_manager.h

```diff
@@ -321,6 +321,12 @@ class ClusterManager {
    * Drain all connection pool connections owned by all clusters in the cluster manager.
    */
   virtual void drainConnections() PURE;
+
+  /**
+   * @param cluster, the cluster to check.
+   * @return true if the cluster is active and static, false otherwise.
+   */
+  virtual bool checkActiveStaticCluster(const std::string& cluster) PURE;
```

Comment:
Could we reuse this feature here? https://github.com/envoyproxy/envoy/blob/main/source/common/grpc/async_client_manager_impl.cc#L49-L57